File: alexandria/ui/src/io/intino/alexandria/ui/displays/components/SelectionOperation.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.notifiers.SelectionOperationNotifier;

import java.util.Collections;
import java.util.List;

public class SelectionOperation<DN extends SelectionOperationNotifier, B extends Box> extends AbstractSelectionOperation<DN, B> implements io.intino.alexandria.ui.displays.components.toolbar.SelectionOperation {
    private List<String> selection = Collections.emptyList();

    public SelectionOperation(B box) {
        super(box);
    }

    public SelectionOperation<DN, B> selection(List<String> selection) {
        this.selection = selection;
        return this;
    }

    public List<String> selection() {
        return selection;
    }

    public void selectedItems(List<String> items) {
        selection(items);
        refresh();
    }

    @Override
    public void refresh() {
        super.refresh();
        notifier.refreshDisabled(selection.size() <= 0);
    }

}

File: alexandria/ui-elements/src/displays/components/SelectionOperation.js
import React from "react";
import AbstractSelectionOperation from "../../../gen/displays/components/AbstractSelectionOperation";
import SelectionOperationNotifier from "../../../gen/displays/notifiers/SelectionOperationNotifier";
import SelectionOperationRequester from "../../../gen/displays/requesters/SelectionOperationRequester";

export default class SelectionOperation extends AbstractSelectionOperation {
	constructor(props) {
		super(props);
		this.notifier = new SelectionOperationNotifier(this);
		this.requester = new SelectionOperationRequester(this);
	};
}